### DISCUSSION TITLE
Why aren't lon and lat being recognized as coordinates?

### DISCUSSION THREAD

**Discussion by kthyng** [xarray-contrib/cf-xarray#352] [Q&A] [answered]
I'd like to be able to read in this model output and be able to recognize lon and lat as coordinates. Can't figure out what is missing.

The example model output file below has both longitude and latitude, which both have `long_name`, `units`, and `standard_name` properly defined. I couldn't find any `coordinates` encoded in any variables that might disrupt either. Any thoughts on what metadata I need on the lon/lat coordinates to get them to be recognized?

```
import xarray as xr

loc = 'https://opendap.co-ops.nos.noaa.gov/thredds/dodsC/NOAA/LOOFS/MODELS/2022/07/21/glofs.loofs.fields.nowcast.20220721.t12z.nc'
ds = xr.open_dataset(loc)
```

**Comment by dcherian** [comment, marked as answer]
I think we require "coordinates" to be in `.coords` It'd be nice to update the docs to be clear on this.
This seems too work
```
ds = xr.open_dataset(loc).cf.set_coords(["latitude", "longitude"])
ds.cf
```

**Reply by kthyng** [reply]
@dcherian Thank you! Yes that does work for me, and I now know two more handy `cf-xarray` functions. Could you say more about how a variable in a netcdf file ends up contained in `.coords`? Is there an attribute I can set somewhere to formalize it instead of running the `.cf.set_coords()` subsequently?

**Reply by dcherian** [reply]
It needs to be listed in a "coordinates" attribute for one of the variables. Try passing `decode_coords="all"` in `open_dataset`

**Reply by dcherian** [reply]
> two more handy cf-xarray functions. 


cf-xarray very blindly tries to wrap all of xarray's API. It's good to just try it. Most of the times it works; if it doesn't we need to explicitly add support.

**Reply by kthyng** [reply]
Right this always get me. The following worked, where I added in coordinates encoding to a variable.
```
loc = 'https://opendap.co-ops.nos.noaa.gov/thredds/dodsC/NOAA/LOOFS/MODELS/2022/07/21/glofs.loofs.fields.nowcast.20220721.t12z.nc'
ds = xr.open_dataset(loc)
ds['temp'].encoding['coordinates'] = 'time sigma lat lon'
ds.isel(time=0).to_netcdf('test.nc')
ds2 = xr.open_dataset('test.nc')
```
though I am pretty sure that coordinates encoding is not always needed because I have some preprocessing code that removes coordinates sometimes in order to get cf-xarray to work properly. I find the "coordinates" use tricky since it is hidden in xarray unless you know to look for it.
Thanks for your help!

**Reply by dcherian** [reply]
> t removes coordinates sometimes in order to get cf-xarray to work properly. 

It's necessary for xarray to do the `set_coords` automatically when it reads in data.
